### PR TITLE
DOCS-57: Fix position & positionCaseInsensitive & positionUTF8 & positionCaseInsensitiveUTF8 functions

### PR DESCRIPTION
**Pull request by godfreyd**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):

- Documentation

Fix position & positionCaseInsensitive functions

### CODE REVIEW

File: docs/en/query_language/functions/string_search_functions.md [hunk 1]
```diff
@@ -2,12 +2,89 @@
 
 The search is case-sensitive by default in all these functions. There are separate variants for case insensitive search.
 
-## position(haystack, needle), locate(haystack, needle)
+## position(haystack, needle) {#position}
 
-Search for the substring `needle` in the string `haystack`.
-Returns the position (in bytes) of the found substring, starting from 1, or returns 0 if the substring was not found.
+Returns the position (in bytes) of the found substring `needle` in the string `haystack`, starting from 1. Returns 0 if the substring was not found.
```
Comment (blinkov) [inline docs/en/query_language/functions/string_search_functions.md]: why do you repeat parameters and return values twice? (here and below)

File: docs/en/query_language/functions/string_search_functions.md [hunk 2]
```diff
@@ -2,12 +2,89 @@
...
 
-For a case-insensitive search, use the function `positionCaseInsensitive`.
+Works under the assumption that the string contains a set of bytes representing a single-byte encoded text. If this assumption is not met, it returns some result.
```
Comment (blinkov) [inline docs/en/query_language/functions/string_search_functions.md]: both strings?

File: docs/en/query_language/functions/string_search_functions.md [hunk 3]
```diff
@@ -2,12 +2,89 @@
...
+
+```text
+┌─position('Hello, world!', 'Hello')─┐
+│                                  1 │
+└────────────────────────────────────┘
+```
+
+## positionCaseInsensitive(haystack, needle) {#positioncaseinsensitive}
+
+Returns the position (in bytes) of the found substring `needle` in the string `haystack`, starting from 1. Returns 0 if the substring was not found. Use the function for a case-insensitive search.
```
Comment (blinkov) [inline docs/en/query_language/functions/string_search_functions.md]: * Do we really need this copy-paste?
* What exactly "case-insensitive search" means? It's important to specify if it works only for English letters in ASCII or more than that.

File: docs/en/query_language/functions/string_search_functions.md [hunk 4]
```diff
@@ -2,12 +2,89 @@
...
+
+**Syntax**
+
+```sql
+positionCaseInsensitive(haystack, needle);
+```
+
+**Parameters**
+
+- `haystack` — string, from which substring is to be searched.
```
Comment (blinkov) [inline docs/en/query_language/functions/string_search_functions.md]:
```suggestion
- `haystack` — string, in which substring will to be searched.
```

File: docs/en/query_language/functions/string_search_functions.md [hunk 5]
```diff
@@ -2,12 +2,89 @@
...
+- `needle` —  substring, which is to be searched.
```
Comment (blinkov) [inline docs/en/query_language/functions/string_search_functions.md]:
```suggestion
- `needle` —  substring, which is to be searched.
```
```suggestion
- `needle` —  substring to be searched.
```
P.S. Afaik long dash (—) is not used in English like it is in Russian.

File: docs/en/query_language/functions/string_search_functions.md [hunk 6]
```diff
@@ -2,12 +2,89 @@
...
+
+**Returned values**
+
+- Starting position (in bytes), if substring was found.
```
Comment (blinkov) [inline docs/en/query_language/functions/string_search_functions.md]: important to mention that it starts with 1, not 0

File: docs/en/query_language/functions/string_search_functions.md [hunk 7]
```diff
@@ -2,12 +2,89 @@
...
+- 0, if the substring was not found.
+
+Type: `Integer`.
```
Comment (blinkov) [inline docs/en/query_language/functions/string_search_functions.md]: Which exactly kind of integers?

File: docs/en/query_language/functions/string_search_functions.md [hunk 8]
```diff
@@ -2,12 +2,89 @@
...
+## position(haystack, needle) {#position}
 
-Search for the substring `needle` in the string `haystack`.
-Returns the position (in bytes) of the found substring, starting from 1, or returns 0 if the substring was not found.
+Returns the position (in bytes) of the found substring `needle` in the string `haystack`, starting from 1. Returns 0 if the substring was not found.
 
-For a case-insensitive search, use the function `positionCaseInsensitive`.
+Works under the assumption that the string contains a set of bytes representing a single-byte encoded text. If this assumption is not met, it returns some result.
+
+For a case-insensitive search, use the function [positionCaseInsensitive](#positioncaseinsensitive).
```
Comment (blinkov) [inline docs/en/query_language/functions/string_search_functions.md]: Why do you mention only CaseInsensitive version, but not UTF8?

**Review by blinkov — CHANGES_REQUESTED** [review]

File: docs/en/query_language/functions/string_search_functions.md (hunk 3 above)
Comment (godfreyd) [reply docs/en/query_language/functions/string_search_functions.md]: > Do we really need this copy-paste?
- Yes.

> What exactly "case-insensitive search" means? It's important to specify if it works only for English letters in ASCII or more than that.

SELECT positionCaseInsensitive('Привет, мир!', 'мир')

┌─positionCaseInsensitive('Привет, мир!', 'мир')─┐
│                                             15 │
└────────────────────────────────────────────────┘

File: docs/en/query_language/functions/string_search_functions.md (hunk 1 above)
Comment (godfreyd) [reply docs/en/query_language/functions/string_search_functions.md]: Removed

File: docs/en/query_language/functions/string_search_functions.md (hunk 4 above)
Comment (godfreyd) [reply docs/en/query_language/functions/string_search_functions.md]: Fixed.

File: docs/en/query_language/functions/string_search_functions.md (hunk 5 above)
Comment (godfreyd) [reply docs/en/query_language/functions/string_search_functions.md]: Fixed

File: docs/en/query_language/functions/string_search_functions.md (hunk 6 above)
Comment (godfreyd) [reply docs/en/query_language/functions/string_search_functions.md]: Fixed

File: docs/en/query_language/functions/string_search_functions.md (hunk 7 above)
Comment (godfreyd) [reply docs/en/query_language/functions/string_search_functions.md]: Yes

File: docs/en/query_language/functions/string_search_functions.md (hunk 8 above)
Comment (godfreyd) [reply docs/en/query_language/functions/string_search_functions.md]: It hasn't been described yet

File: docs/en/query_language/functions/string_search_functions.md (hunk 2 above)
Comment (godfreyd) [reply docs/en/query_language/functions/string_search_functions.md]: What?

File: docs/en/query_language/functions/string_search_functions.md (hunk 5 above)
Comment (godfreyd) [reply docs/en/query_language/functions/string_search_functions.md]: I checked it with linguists. A long dash is used in English. Correct the long dash or a colon.

File: docs/en/query_language/functions/string_search_functions.md (hunk 2 above)
Comment (blinkov) [reply docs/en/query_language/functions/string_search_functions.md]: both arguments

File: docs/en/query_language/functions/string_search_functions.md (hunk 3 above)
Comment (blinkov) [reply docs/en/query_language/functions/string_search_functions.md]: example is not enough, you need to explain how it works

File: docs/en/query_language/functions/string_search_functions.md (hunk 7 above)
Comment (blinkov) [reply docs/en/query_language/functions/string_search_functions.md]: Yes?

File: docs/en/query_language/functions/string_search_functions.md (hunk 8 above)
Comment (blinkov) [reply docs/en/query_language/functions/string_search_functions.md]: why not?

Comment (godfreyd) [reply docs/en/query_language/functions/string_search_functions.md]: already described, see files

File: docs/en/query_language/functions/string_search_functions.md (hunk 7 above)
Comment (godfreyd) [reply docs/en/query_language/functions/string_search_functions.md]: What type can still return?

File: docs/en/query_language/functions/string_search_functions.md (hunk 3 above)
Comment (godfreyd) [reply docs/en/query_language/functions/string_search_functions.md]: Write, example, please

**Pull request closed without merging**